hs-bindgen/src-internal/HsBindgen/Hs/Translation.hs

```diff
@@ -1042,7 +1042,9 @@ functionDecs ::
 functionDecs mu typedefs info f _spec =
     [ Hs.DeclInlineCInclude $ getCHeaderIncludePath $ C.declHeader info
     , Hs.DeclInlineC $ PC.prettyDecl (wrapperDecl innerName wrapperName res args) ""
-    , Hs.DeclForeignImport $ Hs.ForeignImportDecl
+    ] ++
+    toList (Hs.DeclSimple <$> ioComment) ++
```

Comment:
This is I think my only "real" concern with the PR as it stands: the generated header does not make it clear _which_ particular function is "C pure". We should attach it to the function definition itself. But perhaps it would make more sense to fix that after @bolt12 is done with the haddock generation, as it should slot in nicely there.